[nicolaskjh] iP

nicolaskjh opened the pull request:
### LeBron 👑

> "BOOM BOOM BOOM BOOM!" - LeBron James

The **GOAT** of task managers 🐐

- text based
- easy to learn
- fast to use

Let the **GOAT** help you manage your life:

1. Download the app [here](https://github.com/nicolaskjh/ip/releases/tag/A-Jar)
2. Launch the app by double-clicking it
3. Add your tasks
4. Let the GOAT handle the rest.

Features:

- [x] Managing tasks
- [ ] Managing deadlines (_coming soon_)
- [ ] Reminders (_coming soon_)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

nicolaskjh committed:
Added greeting and exit

nicolaskjh committed:
Added echo feature

nicolaskjh committed:
Added ability to store and list text

nicolaskjh committed:
Added ability to mark tasks as done and not done

nicolaskjh committed:
Added Todo, Event and Deadline task types

nicolaskjh committed:
Added automated Text UI testing

nicolaskjh committed:
Refactoring command classes

nicolaskjh committed:
Added error handling for invalid command, empty todo description and no task number given to mark/unmark

nicolaskjh committed:
Added deletion of tasks

nicolaskjh committed:
Package source files as cheeto

nicolaskjh committed:
Fix import statements and code styling

nicolaskjh committed:
Add loading and saving tasks from hard disk

nicolaskjh committed:
Fix code styling

nicolaskjh committed:
Merge branch 'branch-level-7'

bipplane commented on src/main/java/cheeto/storage/Storage.java:
```diff
@@ -86,6 +86,7 @@ public void storeTasks(TaskList tasks){
             String fileDirectory = this.filePath.replace(
                     this.filePath.substring(this.filePath.indexOf("/")), "");
             File file = new File(fileDirectory);
```
If an IOException is thrown during file operations, this method should indicate that it throws IOException.

Note: Do also ensure there is a space between `(TaskList tasks)` and `{` in the method signature.

bipplane commented on src/main/java/cheeto/command/DeleteCommand.java:
```diff
@@ -14,6 +14,7 @@ public DeleteCommand(int idx) {
     public String getResponse(TaskList taskList) {
         Task removed = taskList.getTask(idx);
         taskList.removeTask(this.idx);
```
Be careful! It doesn't seem like you're checking if the `idx` given by the user is a valid number to begin with. This could lead to an unwanted `ArrayIndexOutOfBoundsException` that isn't getting handled (yet)

bipplane commented on data/cheeto.txt:
```diff
@@ -3,4 +3,3 @@ D, 0, return book, June 6th
 D, 1, project meeting, Aug 6th 2pm
 T, 1, join sports club
 T, 0, meow
-D, 1, meow, Jan 31st
```
This is pretty cute! ❤️

bipplane reviewed (commented):
Great work so far! Keep it up! >:3

nicolaskjh committed:
Add date parser

nicolaskjh committed:
Add date time parsing from input

nicolaskjh committed:
Fix input file format

nicolaskjh committed:
Merge branch 'master' into branch-Level-8

nicolaskjh committed:
Fix text inputs

nicolaskjh committed:
Fix date time parsing in storage

nicolaskjh committed:
Refactor to include OOP

nicolaskjh committed:
Merge branch 'branch-More-OOP' into master

nicolaskjh committed:
Add classes into packages

nicolaskjh committed:
Merge branch 'add-gradle-support' into master

nicolaskjh committed:
Add gradle support

nicolaskjh committed:
Add unit tests for input parser

nicolaskjh committed:
Package app as jar file

nickt121 commented on src/main/java/lebron/task/Task.java:
```diff
@@ -0,0 +1,40 @@
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
```
I like how you followed the naming scheme for boolean variables.

nickt121 commented on src/main/java/lebron/task/Deadline.java:
```diff
@@ -0,0 +1,25 @@
+
+    @Override
+    public String toDataString() {
+        return "D, " + super.getStatusData() + ", " + super.getDescription()
```
Perhaps using a format string here might make the code a little cleaner

nickt121 commented on src/main/java/lebron/parser/InputParser.java:
```diff
@@ -0,0 +1,119 @@
+        return new UnmarkDoneCommand(Integer.parseInt(split[1]) - 1);
+    }
+
+    public static Command readDeleteInput(String input) throws LebronException {
```
It seems like this method is very similar to the one above. Perhaps you can try developing a method that encompasses both functions.

nickt121 commented on src/main/java/lebron/storage/Storage.java:
```diff
@@ -0,0 +1,104 @@
+public class Storage {
+    private final String filePath;
+
+    private final String TASK_TODO = "T";
```
Perhaps you can use an enum for this part, especially if more types of tasks are to be expected in the future.

nickt121 commented on src/main/java/lebron/command/MarkDoneCommand.java:
```diff
@@ -0,0 +1,19 @@
+import lebron.task.TaskList;
+
+public class MarkDoneCommand extends Command {
+    private int idx;
```
Perhaps a more intuitive variable name here?

nickt121 reviewed (commented):
Great job so far. Keep up the good work!!

wilsonh0 commented on src/main/java/lebron/command/DeleteCommand.java:
```diff
@@ -0,0 +1,21 @@
+
+    public DeleteCommand(int idx) {
+        this.idx = idx;
+    }
```
Could try renaming idx for better readability.

wilsonh0 commented on src/main/java/lebron/command/Command.java:
```diff
@@ -0,0 +1,11 @@
+    public boolean isExit() {
+        return false;
+    }
+}
```
I like your use of the Command Pattern! It helps encapsulate the logic for different actions.

wilsonh0 commented on src/main/java/lebron/command/UnmarkDoneCommand.java:
```diff
@@ -0,0 +1,19 @@
+    public UnmarkDoneCommand(int idx) {
+        this.idx = idx;
+    }
+
```
Similarly here, idx can be renamed for better readability.

wilsonh0 commented on src/main/java/lebron/Lebron.java:
```diff
@@ -0,0 +1,54 @@
+    public void exit() {
+        storage.storeTasks(taskList);
+    }
+}
```
I noticed that the classes and methods are missing Javadocs. Adding them would improve readability and maintainability.

wilsonh0 reviewed (commented):
I like that you use a ton of separations between logical components, which makes the code more readable and easier to follow. Overall, the code quality is good and follows coding standards well. Only thing missing are javadocs.

nicolaskjh committed:
Add JavaDocs for all classes and methods

nicolaskjh committed:
Change idx variable names

nicolaskjh committed:
Add find tasks functionality

nicolaskjh committed:
Add find tasks functionality

nicolaskjh committed:
Merge branch 'branch-A-JavaDoc' into master

nicolaskjh committed:
Merge branch 'branch-A-CodingStandard' into master

nicolaskjh committed:
Merge branch 'branch-Level-9' into master

nicolaskjh committed:
Fix coding style errors

nicolaskjh committed:
Fix errors detected by checkstyle

nicolaskjh committed:
Merge branch 'branch-A-CheckStyle' into master

nicolaskjh committed:
Add dependencies for JavaFX

nicolaskjh committed:
Add GUI

nicolaskjh committed:
Add new chatbot responses

nicolaskjh committed:
Add GUI elements

nicolaskjh committed:
Add GUI elements

nicolaskjh committed:
Add assertions to InputParser class

The Mark, Unmark and Delete commands require a task number to be
entered. The task number needs to exist in the current Task List.

Adding assertions to assert that the task number is a valid task number
ensures that we do not run into IndexOutOfBounds error when accessing
the task from the task list.

nicolaskjh committed:
Remove case-sensitive search for FindCommand

FindCommand is case-sensitive.

A case-insensitive find is more user friendly as users cannot be
expected to remember the exact case of the keywords.

Let's convert the keyword and task description to lower case in
the filterTasks() method in TaskList to allow case-insensitive
filtering of tasks.

nicolaskjh committed:
Change toString methods to use String format

nicolaskjh committed:
Merge pull request #1 from nicolaskjh/branch-A-Assertions

Add assertions to InputParser

nicolaskjh committed:
Merge branch 'master' into branch-A-CodeQuality

nicolaskjh committed:
Merge pull request #2 from nicolaskjh/branch-A-CodeQuality

Change toString methods to improve code quality

nicolaskjh committed:
Add task priority

nicolaskjh committed:
Merge branch 'branch-C-priority' into master

nicolaskjh committed:
Fix indexing error in Storage

nicolaskjh committed:
Change task priority tag

nicolaskjh committed:
Remove unused function in DateParser

nicolaskjh committed:
Add error handling to storeTasks

nicolaskjh committed:
Add GUI changes

nicolaskjh committed:
Refactor InputParser

nicolaskjh committed:
Update README.md

nicolaskjh committed:
Update README.md

nicolaskjh committed:
Add GUI screenshot to README.md

nicolaskjh committed:
Update README.md

nicolaskjh committed:
Change import order in InputParser
